tests/frame_test.go

```diff
@@ -21,3 +23,58 @@ func TestFramePress(t *testing.T) {
 
 	require.Equal(t, "AbC", f.InputValue("#text1", nil))
 }
+
+func TestFrameDismissDialogBox(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "alert",
+		},
+		{
+			name: "confirm",
+		},
+		{
+			name: "prompt",
+		},
+		{
+			name: "beforeunload",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			b := newTestBrowser(t, withFileServer())
+
+			p := b.NewPage(nil)
+
+			err := b.await(func() error {
+				opts := b.toGojaValue(struct {
+					WaitUntil string `js:"waitUntil"`
+				}{
+					WaitUntil: "networkidle",
+				})
+				b.promise(p.Goto(b.staticURL("dialog.html?dialogType="+tt.name), opts)).then(func() *goja.Promise {
```

Comment:
Consider (optional) to make it slightly more readable and comprehensible.

```suggestion
				opts := b.toGojaValue(struct {
					WaitUntil string `js:"waitUntil"`
				}{
					WaitUntil: "networkidle",
				})
				pageGoto := p.Goto(
					b.staticURL("dialog.html?dialogType="+tt.name),
					opts,
				)
				b.promise(pageGoto).then(func() *goja.Promise {
```